lib/resources/bankAccounts.js

```diff
@@ -1,107 +1,43 @@
-var request = require('request');
-var BankAccounts;
+'use strict';
 
-function BankAccounts (config) {
-  this.uri = config.endpoint + 'bank_accounts';
-  this.apiKey = config.apiKey;
-  this.userAgent = config.userAgent;
-  this.apiVersion = config.apiVersion;
-  return this;
-}
+var util         = require('util');
+var ResourceBase = require('./resourceBase');
 
-BankAccounts.prototype.retrieve = function (id, done) {
-  request(
-    {
-      url: this.uri + '/' + id,
-      auth: {
-        user: this.apiKey,
-        password: ''
-      },
-      headers: this.headers,
-      json: true,
-      method: 'GET'
-    }, function (e, r, body) {
-      if (body && body.errors) {
-        done(body.errors, body);
-      } else {
-        done(e, body);
-      }
-    });
+var BankAccounts = function(config) {
+  ResourceBase.call(this, 'bank_accounts', config);
 };
 
-BankAccounts.prototype.delete = function (id, done) {
-  request(
-    {
-      url: this.uri + '/' + id,
-      auth: {
-        user: this.apiKey,
-        password: ''
-      },
-      headers: this.headers,
-      json: true,
-      method: 'DELETE'
-    }, function (e, r, body) {
-      if (body && body.errors) {
-        done(body.errors, body);
-      } else {
-        done(e, body);
-      }
-    });
-};
+util.inherits(BankAccounts, ResourceBase);
 
-BankAccounts.prototype.list = function (options, done) {
-  var count;
-  var offset;
-  if (typeof options === 'function') {
-    count = 10;
-    offset = 0;
-    done = options;
-  } else {
-    count = options.count || 10;
-    offset = options.offset || 0;
-  }
-  request(
-    {
-      url: this.uri,
-      auth: {
-        user: this.apiKey,
-        password: ''
-      },
-      qs: {
-        count: count,
-        offset: offset
-      },
-      headers: this.headers,
-      json: true,
-      method: 'GET'
-    }, function (e, r, body) {
-      if (body && body.errors) {
-        done(body.errors, body);
-      } else {
-        done(e, body);
-      }
-    });
-};
+(function() {
 
-BankAccounts.prototype.create = function (params, done) {
-  request(
-    {
-      url: this.uri,
-      auth: {
-        user: this.apiKey,
-        password: ''
-      },
-      headers: this.headers,
-      json: true,
-      form: params,
-      method: 'POST'
-    }, function (e, r, body) {
-      if (body && body.errors) {
-        done(body.errors, body);
-      } else {
-        done(e, body);
-      }
-    });
-};
+  this.list = function(options, callback) {
+
+    if(typeof options === 'function') {
+      callback = options;
+      options  = {};
+    }
+
+    options = {
+      count:  options.count  || 10,
```

Comment:
No - he's just saying that the API will default to 10 for count and 0 for offset so there is no need to set the defaults in the wrapper.